overlord/snapshotstate/snapshotstate.go

```diff
@@ -293,7 +293,7 @@ func checkSnapshotTaskConflict(st *state.State, setID uint64, conflictingKinds .
 var List = backend.List
 
 // Import a given snapshot ID from an exported snapshot
-func Import(ctx context.Context, st *state.State, r io.Reader) (uint64, []string, int64, error) {
+func Import(ctx context.Context, st *state.State, r io.Reader, size int64) (uint64, []string, int64, error) {
```

Comment:
maybe s/size/advertisedSize/ or sizeHint? since it's basically a claim made by the client?

Reply:
It's more than a hint, we will use this to validate e.g. disk-space, if we can't trust it it's useless. Or am I misunderstanding?

Reply:
Yeah, I think we covered this on IRC. Even if there's a small zip content followed by garbage, this is still the top limit of we are going to store, so in that sense it's good.